lib/msf/core/post/linux/compile.rb

```diff
@@ -11,35 +11,43 @@ def initialize(info = {})
     super
     register_options( [
       OptEnum.new('COMPILE', [true, 'Compile on target', 'Auto', ['Auto', 'True', 'False']]),
+      OptEnum.new('COMPILER', [true, 'Compiler to use on target', 'gcc', ['gcc', 'clang']]),
     ], self.class)
   end
 
   def live_compile?
     return false unless %w{ Auto True }.include?(datastore['COMPILE'])
 
-    if has_gcc?
-      vprint_good 'gcc is installed'
-      return true
+    if datastore['COMPILER'] == gcc
+      if has_gcc?
+        vprint_good 'gcc is installed'
+        return true
+      end
+    elsif datastore['COMPILER'] == clang
+      if has_clang?
+        vprint_good 'clang is installed'
+        return true
+      end
     end
 
     unless datastore['COMPILE'] == 'Auto'
-      fail_with Module::Failure::BadConfig, 'gcc is not installed. Set COMPILE False to upload a pre-compiled executable.'
+      fail_with Module::Failure::BadConfig, "#{datastore['COMPILER']} is not installed. Set COMPILE False to upload a pre-compiled executable."
     end
   end
```

Comment:
Ruby semantics dictate that a method ending in a question mark should return true or false so this would be a good opportunity to adjust the return type here to follow that guidance.

```suggestion
    end
    
    false
  end
```